src/libraries/System.Private.Uri/src/System/DomainNameHelper.cs

```diff
@@ -200,10 +195,14 @@ public static string IdnEquivalent(string hostname)
         {
             // check if only ascii chars
             // special case since idnmapping will not lowercase if only ascii present
-            if (IsAscii(hostname))
+            if (Ascii.IsValid(hostname))
             {
                 // just lowercase for ascii
-                return hostname.ToLowerInvariant();
+                return string.Create(hostname.Length, hostname, static (chars, asciiHostName) =>
+                {
+                    OperationStatus status = Ascii.ToLower(asciiHostName, chars, out _);
+                    Debug.Assert(status == OperationStatus.Done);
+                });
```

Comment:
```suggestion
                return hostname.ToLowerInvariant();
```

The common case here is that the `ToLowerInvariant` will see the input is already lowercase and return the original instance. We don't want the additional allocation.